hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConsts.java

```diff
@@ -434,6 +434,10 @@ private OzoneConsts() {
   public static final Pattern KEYNAME_ILLEGAL_CHARACTER_CHECK_REGEX  =
           Pattern.compile("^[^\\\\{}<>^%~#|`\\[\\]\"\\x80-\\xff]+$");
 
+  // Supported characters for resource names when isStrictS3 is set as false
+  public static final Pattern SUPPORTED_CHARACTER_CHECK_REGEX  =
+      Pattern.compile("[a-z0-9.-]");
```

Comment:
I was suspicious about the final dash ( "-" )  character in the regex as the dash character is used for ranges inside the character classes, and I thought it needed escaped. However I tested a few regexes and it seems to work fine, and if you have an invalid range the regex fails to compile. I am just leaving this comment here incase someone else thinks to check this.

Reply:
Indeed.

In theory we could check the whole name string **once** with the regex (with an extra `+` at the end of course) in `verifyResourceName` to avoid checking **each** character one by one. But that would require some refactoring on the whole resource name checking logic, and need to avoid disrupting the functionality of the `isStrictS3` config.

@Tejaskriya Can we do a quick unit test benchmark of the `isSupportedCharacter` method, e.g. run it over a 1 million characters and compare with the old checker, to see if it is indeed slower? And if so, by how much.